lib/floe/workflow/runner/kubernetes.rb

```diff
@@ -92,6 +92,21 @@ def output(runner_context)
             end
         end
 
+        def wait(contexts = [], timeout: nil, since: Time.now.utc, events: [], &block)
+          refs = Array(contexts).map { |ctx| ctx["container_ref"] }
+
+          kubeclient.watch_pods(:namespace => namespace) do |notice|
+            ref = notice.object.metadata.name
+            next unless refs.empty? || refs.include?(ref)
+
+            if block
+              yield ref
+            else
+              return ref
+            end
+          end
+        end
```

Comment:
Since we're not actually using the block here, I think it's a little more performant to not capture it into a variable.

```suggestion
        def wait(contexts = [], timeout: nil, since: Time.now.utc, events: [])
          refs = Array(contexts).map { |ctx| ctx["container_ref"] }

          kubeclient.watch_pods(:namespace => namespace) do |notice|
            ref = notice.object.metadata.name
            next unless refs.empty? || refs.include?(ref)

            if block_given?
              yield ref
            else
              return ref
            end
          end
        end
```

Reply:
Unless you're trying to keep it consistent with the other runner, which is fine too